### PR TITLE
Automated release

### PR DESCRIPTION
**Pull request by benthomasson**
Adds an automated release process using ansible and github actions.

Run locally with:
```
ansible-playbook tools/ansible/release.yml -i localhost -e github_token=${GH_TOKEN}
```

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+jobs:
+  stage:
+    runs-on: ubuntu-latest
+    timeout-minutes: 90
+    permissions:
+      packages: write
+      contents: write
+    steps:
+      - name: Checkout dab
```
Comment (Dostonbek1) [inline .github/workflows/release.yml:18]: same comment on "dab" as in ansible-rulebook

**Review by Dostonbek1 — APPROVED** [review]

**Review by Dostonbek1 — APPROVED** [review]

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    timeout-minutes: 90
+    permissions:
+      packages: write
+      contents: write
+    steps:
+      - name: Checkout eda-server
+        uses: actions/checkout@v3
+
+      - name: Get python version from Makefile
+        run: echo py_version=`make PYTHON_VERSION` >> $GITHUB_ENV
```
Comment (Alex-Izquierdo) [inline .github/workflows/release.yml:22]: This line would not going to fail? I don't see any PYTHON_VERSION in the makefile.

Comment (benthomasson) [reply .github/workflows/release.yml:22]: I am still trying to find a way to test this.

**Comment by Alex-Izquierdo** [comment]
In this project we are using `task` I think there is no need to duplicate tools adding a makefile. 
https://github.com/ansible/eda-server/blob/main/Taskfile.dist.yaml

File: .github/workflows/release.yml (hunk 2 above)
Comment (benthomasson) [reply .github/workflows/release.yml:22]: I added this make target

**Comment by benthomasson** [comment]
> In this project we are using `task` I think there is no need to duplicate tools adding a makefile. https://github.com/ansible/eda-server/blob/main/Taskfile.dist.yaml

Task is not available everywhere and is actually hard to install on some older versions of Fedora.     We should consider using makefile since it is the standard on the rest of the platform.

**Comment by benthomasson** [comment]
I removed the Makefile and used task instead.

File: Taskfile.dist.yaml [hunk 3]
```diff
@@ -355,3 +355,16 @@ tasks:
       - task: macpodman
         vars:
           CLI_ARGS: tunnel
+
+  build:release:
+    desc: "builds all the files needed for a release"
+    cmds:
```
Comment (Alex-Izquierdo) [inline Taskfile.dist.yaml:361]: Is missing the build command?

File: pyproject.toml [hunk 4]
```diff
@@ -120,3 +120,25 @@ ban-relative-imports = "parents"
 
 [tool.ruff.pydocstyle]
 convention = "pep257"
+
+[tool.setuptools_scm]
```
Comment (Alex-Izquierdo) [inline pyproject.toml:124]: should we use poetry for building as it is configured? 
https://github.com/ansible/eda-server/blob/main/pyproject.toml#L3

**Comment by benthomasson** [comment]
Consumers of eda-server should use the latest build from quay.io

**Pull request closed without merging**